### PR TITLE
Share api response fix

### PR DESCRIPTION
**Pull request by Prajwal17Tunerlabs**
# Description
These recommendations are intended to promote code quality and team communication during software development. They cover a variety of topics, including ensuring that pull requests are submitted to the correct branch, documenting new methods, preserving consistency across many services, and avoiding typical blunders like accessing APIs or DB queries within loops. Sensitive data should not be uploaded, and changes to environment variables or database models should be executed consistently. Teams may work more effectively and develop higher-quality software by adhering to these standards.

## Type of change
Please choose appropriate options.

- [ ]  Bug fix (non-breaking change which fixes an issue)
- [ ]  New feature (non-breaking change which adds functionality)
- [ ]  Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ]  Enhancement (additive changes to improve performance)
- [ ]  This change requires a documentation update

## Checklist 

- [ ]  It's critical to avoid making needless file modifications in contributions, such as adding new lines, console logs, or additional spaces, to guarantee cleaner and more efficient code. Furthermore, eliminating unnecessary imports from a file might enhance code readability and efficiency.
- [ ]  Ensure that the pull request is assigned to the right base branch and that the development branch name contains the JIRA Task Id. Furthermore, each commit message should include the JIRA Task Id in the manner "ED-100: message".
- [ ]  Only update packages if it is mentioned and authorized in the design document, and make sure that you have the required permissions.
- [ ]  Avoid making API and database queries inside a loop as it can lead to performance issues and slow down the system.
- [ ]  When calling another function inside a given function, add comments explaining the purpose and meaning of the passed arguments and expected return values.
- [ ]  If adding a blank argument in a function, add a comment explaining the reason for the blank argument.
- [ ]  Before submitting a pull request, do a self-review of your code to ensure there are no conflicts with the base branch and all comments have been addressed.
- [ ]  Before merging a pull request, it's important to have other team members review it to catch any potential errors or issues
- [ ]  To maintain code integrity, it's important to remove all related changes when removing code during a code review.
- [ ]  If new constants, endpoints, or utility functions are introduced, it is important to check if they already exist in the service to avoid any duplication.
- [ ]  Whenever a new environment variable is added to a service, it's important to ensure that the necessary changes are made to related files such as ".env.sample" and "envVariables.js" to maintain consistency and avoid errors. Additionally, the new environment variable should be added to the devops repository to ensure that it is properly documented and accessible to the team.
- [ ]  When adding a new function to a service, it is important to document it with relevant information such as the name, parameters, and return value in a consistent format across all services. Additionally, if there are any changes to the API response, ensure that the documentation in the controllers is updated accordingly. 
- [ ]  Write a clear and concise commit message that describes the changes made.
- [ ]  Maintain consistent function signature and code across all services when adding a function to multiple services. Implement changes to database models in all services that use the same model. 
- [ ]  Use only let and const. Do not use var.
- [ ]  Make common functions for repetitive code blocks.
- [ ]  Avoid uploading sensitive information such as secret tokens or passwords in pull requests to ensure data security.
- [ ]  Maintain consistent indentation and spacing throughout the code.

### CODE REVIEW

File: controllers/v1/cloud-services/files.js [hunk 1]
```diff
@@ -156,4 +156,45 @@ module.exports = class Files {
...
+
+	/**
+	 * @api {get} /project/v1/cloud-services/files/download
+	 * @apiVersion 1.0.0
+	 * @apiHeader {String} X-auth-token Authenticity token
+	 * @apiSampleRequest /project/v1/cloud-services/files/download?file=project/77d9b0a7-f962-4f24-9fbd-cb027ada5eee/1/187aa81b-3007-4122-b5f0-ea0cc70af2fd/c56938e3-26aa-4a69-a8ef-9eaceeb0ca1b.pdf
+	 * @apiUse successBody
+	 * @apiUse errorBody
+	 * @apiParamExample {json} Response:
+	 * Automaticaly downloads the file.
```
Comment (borkarsaish65) [inline controllers/v1/cloud-services/files.js:168]: You can write directly server the file as a response

File: generics/helpers/common_handler.js [hunk 2]
```diff
@@ -117,63 +117,23 @@ const improvementProjectPdfGeneration = async function (responseData, userId) {
...
+										})
 									} else {
 										return resolve({
-											status: CONSTANTS.common.STATUS_FAILURE,
-											message: pdfDownloadableUrl.message
-												? pdfDownloadableUrl.message
-												: CONSTANTS.common.COULD_NOT_GENERATE_PDF,
+											status: false,
+											message: CONSTANTS.common.COULD_NOT_GENERATE_PDF,
 											pdfUrl: '',
```
Comment (borkarsaish65) [inline generics/helpers/common_handler.js:137]: is this pdfUrl key required ? if the response is already a failure ? can it not work without pdfUrl key not being present ?

File: module/userProjects/helper.js [hunk 3]
```diff
@@ -2022,18 +2031,19 @@ module.exports = class UserProjectsHelper {
...
 				}
 
-				if (response && response.success) {
+				if (response && response.success && response.folderPath != '') {
+					const downloadUrl = `${process.env.ELEVATE_PROJECT_SERVICE_URL}/${process.env.SERVICE_NAME}${CONSTANTS.endpoints.AUTO_DOWNLOAD}?file=${response.folderPath}`
 					return resolve({
 						success: true,
 						message: CONSTANTS.apiResponses.REPORT_GENERATED_SUCCESSFULLY,
 						data: {
 							data: {
```
Comment (borkarsaish65) [inline module/userProjects/helper.js:2040]: why have you nested the url in data.data ? can you avoid doing that ?
also result key is used same way url is nested in data.data.
Why is it used twice ?

File: controllers/v1/cloud-services/files.js (hunk 1 above)
Comment (Prajwal17Tunerlabs) [reply controllers/v1/cloud-services/files.js:168]: updated the api doc.

**Pull request merged**